src/osd/ReplicatedBackend.cc

```diff
@@ -1316,6 +1316,12 @@ void ReplicatedBackend::calc_head_subsets(
   uint64_t size = obc->obs.oi.size;
   if (size)
     data_subset.insert(0, size);
+  map<hobject_t, pg_missing_t::item, hobject_t::ComparatorWithDefault>::const_iterator it
+    = missing.missing.find(head);
+  assert(it != missing.missing.end());
+  can_recover_partial = it->second.can_recover_partial;
+  if (can_recover_partial)
+    data_subset.intersection_of(it->second.dirty_extents);
```

Comment:
if can_recover_partial, use dirty_extents, then can't use "data_subset.subtract(cloning)" in the following, because dirty_extents is none of business with clone_verlap